### ISSUE TITLE
CI: GKE: K8sUpdates Tests upgrade and downgrade from a Cilium stable image to master

### ISSUE THREAD

**Issue by pchaigno** [cilium/cilium#13276]
This test is currently failing about 1/4 on GKE and was quarantined about a month ago. The current error is displayed below but it may be hiding other issues given this has been quarantined for a while, in https://github.com/cilium/cilium/commit/fef1a0d4f6f95e7843a25a9eeb439482aa95156a. It wasn't documented at the time, so there is no way to know what was the error back then.

https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline-GKE-Quarantine/102/testReport/junit/Suite-k8s-1/15/K8sUpdates_Tests_upgrade_and_downgrade_from_a_Cilium_stable_image_to_master/
[1515c6eb_K8sUpdates_Tests_upgrade_and_downgrade_from_a_Cilium_stable_image_to_master.zip](https://github.com/cilium/cilium/files/5276507/1515c6eb_K8sUpdates_Tests_upgrade_and_downgrade_from_a_Cilium_stable_image_to_master.zip)

https://datastudio.google.com/s/qUoyP7yAj_I

### Stacktrace
```
/home/jenkins/workspace/Ginkgo-CI-Tests-Pipeline-GKE-Quarantine/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:514
Cilium "1.8.90" was not able to be deleted
Expected command: helm delete cilium --namespace=cilium 
To succeed, but it failed:
Exitcode: -1 
Err: signal: killed
Stdout:
 	 
Stderr:
 	 

/home/jenkins/workspace/Ginkgo-CI-Tests-Pipeline-GKE-Quarantine/src/github.com/cilium/cilium/test/k8sT/Updates.go:256
```

I'm opening this to track progress on the fix.

**Comment by pchaigno** [comment]
It looks like that issue was just replaced by a new one on K8s 1.16 :cry: 
```
/home/jenkins/workspace/Ginkgo-CI-Tests-Pipeline-GKE-Quarantine/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:514
Cilium "1.8.90" did not become ready in time
Expected
    <*errors.errorString | 0xc001091da0>: {
        s: "4m0s timeout expired",
    }
to be nil
/home/jenkins/workspace/Ginkgo-CI-Tests-Pipeline-GKE-Quarantine/src/github.com/cilium/cilium/test/k8sT/Updates.go:255
```

**Comment by pchaigno** [comment]
#13793 should fix the K8s 1.16 variant of this flake.
